aptos-move/framework/aptos-framework/sources/confidential_asset/confidential_asset.move

```diff
@@ -0,0 +1,1127 @@
+        token: Object<Metadata>): confidential_balance::CompressedConfidentialBalance acquires ConfidentialAssetStore
+    {
+        assert!(has_confidential_asset_store(owner, token), error::not_found(ECA_STORE_NOT_PUBLISHED));
+
+        let ca_store = borrow_global<ConfidentialAssetStore>(get_user_address(owner, token));
+
+        ca_store.actual_balance
+    }
+
+    #[view]
+    /// Returns the encryption key (EK) of the user for the specified token.
+    public fun encryption_key(
+        user: address,
+        token: Object<Metadata>): twisted_elgamal::CompressedPubkey acquires ConfidentialAssetStore
+    {
+        assert!(has_confidential_asset_store(user, token), error::not_found(ECA_STORE_NOT_PUBLISHED));
+
+        borrow_global_mut<ConfidentialAssetStore>(get_user_address(user, token)).ek
+    }
+
+    #[view]
+    /// Checks if the user's actual balance is normalized for the specified token.
+    public fun is_normalized(user: address, token: Object<Metadata>): bool acquires ConfidentialAssetStore {
+        assert!(has_confidential_asset_store(user, token), error::not_found(ECA_STORE_NOT_PUBLISHED));
+
+        borrow_global<ConfidentialAssetStore>(get_user_address(user, token)).normalized
+    }
+
+    #[view]
+    /// Checks if the user's confidential asset store is frozen for the specified token.
+    public fun is_frozen(user: address, token: Object<Metadata>): bool acquires ConfidentialAssetStore {
+        assert!(has_confidential_asset_store(user, token), error::not_found(ECA_STORE_NOT_PUBLISHED));
+
+        borrow_global<ConfidentialAssetStore>(get_user_address(user, token)).frozen
+    }
+
+    #[view]
+    /// Returns the asset-specific auditor's encryption key.
+    /// If the auditing feature is disabled for the token, the encryption key is set to `None`.
+    public fun get_auditor(
+        token: Object<Metadata>): Option<twisted_elgamal::CompressedPubkey> acquires FAConfig, FAController
+    {
+        let fa_config_address = get_fa_config_address(token);
+
+        if (!is_allow_list_enabled() && !exists<FAConfig>(fa_config_address)) {
+            return std::option::none();
+        };
+
+        borrow_global<FAConfig>(fa_config_address).auditor_ek
+    }
+
+    #[view]
+    /// Returns the circulating supply of the confidential asset.
+    public fun confidential_asset_balance(token: Object<Metadata>): u64 acquires FAController {
+        let fa_store_address = get_fa_store_address();
+        assert!(primary_fungible_store::primary_store_exists(fa_store_address, token), EINTERNAL_ERROR);
+
+        primary_fungible_store::balance(fa_store_address, token)
+    }
+
+    //
+    // Public functions that correspond to the entry functions and don't require serializtion of the input data.
+    // These function can be useful for external contracts that want to integrate with the Confidential Asset protocol.
+    // 
+
+    /// Implementation of the `register` entry function.
+    public fun register_internal(
```

Comment:
no need to be public?

Reply:
same Q for all `_internal` functions in this file

Reply:
Made public so other contracts can call these function without params serialization